test/ck_tile/gemm/gemm_kernel_validation_base.hpp

```diff
@@ -0,0 +1,231 @@
+        const CLayout layout_c{};
+
+        // Calculate strides based on tensor dimensions and layout
+        ck_tile::index_t stride_a = is_row_major(layout_a) ? K : M;
+        ck_tile::index_t stride_b = is_row_major(layout_b) ? N : K;
+        ck_tile::index_t stride_c = is_row_major(layout_c) ? N : M;
+
+        // Create host tensors for test data
+        ck_tile::HostTensor<ADataType> a_m_k(
+            ck_tile::host_tensor_descriptor(M, K, stride_a, is_row_major(layout_a)));
+        ck_tile::HostTensor<BDataType> b_k_n(
+            ck_tile::host_tensor_descriptor(K, N, stride_b, is_row_major(layout_b)));
+        ck_tile::HostTensor<CDataType> c_m_n_dev_result(
+            ck_tile::host_tensor_descriptor(M, N, stride_c, is_row_major(layout_c)));
+        ck_tile::HostTensor<CDataType> c_m_n_host_result(
+            ck_tile::host_tensor_descriptor(M, N, stride_c, is_row_major(layout_c)));
+
+        // Initialize input tensors with random data
+        ck_tile::FillUniformDistribution<ADataType>{-0.5f, 0.5f}(a_m_k);
+        ck_tile::FillUniformDistribution<BDataType>{-0.5f, 0.5f}(b_k_n);
+        c_m_n_dev_result.SetZero();
+
+        // Apply structured sparsity pattern to matrix A if requested
+        if(structured_sparsity)
+        {
+            ck_tile::AdjustToStructuredSparsity<ADataType>{}(a_m_k);
+        }
+
+        // Allocate device memory buffers
+        ck_tile::DeviceMem a_m_k_dev_buf(a_m_k.get_element_space_size_in_bytes());
+        ck_tile::DeviceMem b_k_n_dev_buf(b_k_n.get_element_space_size_in_bytes());
+        ck_tile::DeviceMem c_m_n_dev_buf(c_m_n_dev_result.get_element_space_size_in_bytes());
+
+        // Copy matrix A to device
+        a_m_k_dev_buf.ToDevice(a_m_k.data());
+
+        // Handle special data type conversions for matrix B
+        if constexpr(std::is_same_v<BDataType, ck_tile::pk_int4_t>)
+        {
+            // Apply int4 vector permutation for device compatibility
+            ck_tile::HostTensor<BDataType> b_k_n_dev = b_k_n;
+            permute_vectors_i4x4_b(b_k_n_dev);
+            b_k_n_dev_buf.ToDevice(b_k_n_dev.data());
+        }
+        else
+        {
+            b_k_n_dev_buf.ToDevice(b_k_n.data());
+        }
+
+        // Initialize output buffer to zero
+        c_m_n_dev_buf.SetZero();
+
+        // Prepare kernel arguments structure
+        ck_tile::GemmHostArgs gemm_args = {
+            a_m_k_dev_buf.GetDeviceBuffer(),
+            b_k_n_dev_buf.GetDeviceBuffer(),
+            c_m_n_dev_buf.GetDeviceBuffer(),
+            split_k,
+            M,
+            N,
+            K,
+            stride_a,
+            stride_b,
+            stride_c,
+        };
+
+        // Compute reference result for validation
+        compute_reference_gemm(a_m_k, b_k_n, c_m_n_host_result);
+
+        try
+        {
+            // Configure stream for kernel execution
+            ck_tile::stream_config stream_cfg{nullptr, true, 0, 1, 1, true, false, 1};
```

Comment:
No need for warmup & kernel timing. Just single iteration is enough.

```suggestion
            ck_tile::stream_config stream_cfg{nullptr, false, 0, 0, 1};
```

Reply:
Done